src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -501,7 +513,18 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
 
         var parameterCustomAttributes = parameter.GetCustomAttributes();
 
-        if (parameterCustomAttributes.OfType<IFromRouteMetadata>().FirstOrDefault() is { } routeAttribute)
+        if (parameter.CustomAttributes.Any(a => typeof(ParametersAttribute).IsAssignableFrom(a.AttributeType)) ||
+            parameter.ParameterType.CustomAttributes.Any(a => typeof(ParametersAttribute).IsAssignableFrom(a.AttributeType)))
+        {
+            if (parameter is SurrogateParameterInfo)
+            {
+                throw new NotSupportedException(
+                    $"Nested {nameof(ParametersAttribute)} is not supported and should be used only for handler parameters or parameter types.");
```

Comment:
In my opinion, we should avoid a complex model binding and only bind top level. 

Eg.:

``` c#
[Parameters]
record Person(string Id, string Name, [Parameters] Address Address);
record Address(string Street, [Parameters] City City);
record City(string CityName, string State);
```

``` c#
app.MapPost("/customers/", (Person person) => Results.Created());
```

Reply:
I think we should do top level only as well, but I'm wiling to discuss. Avoiding walking object graphs seems like a good thing but it might be unexpected if you are trying to compose many of these objects.

Reply:
My example is not the best but in my mind the composition of these "argument list objects" is something like this:

``` c#
app.MapPost("/customers/", (Person person, Address address, City city) => Results.Created());

[Parameters]
record Person(string Id, string Name);
[Parameters]
record Address(string Street);
[Parameters]
record City(string CityName, string State);
```

A better example of this could be the one listed in the issue:

``` c#
app.MapGet("/products/{id}", ([Parameters]ProductRequest req, [Parameters]Paging pageInfo) =>
{    
     return req.Db.Products.Skip((req.Page - 1) * req.PageSize).Take(req.PageSize);
});

public record ProductRequest(int Id, ILogger<Program> Logger, MyDb Db);
public record Paging(int PageSize, int Page);
```

I am open to change it and if we decide to walk the object graph, the current binding logic works fine (of course we will need to remove the condition that throws) but I will need to quick update in how the APIExplore/OpenAPI works and might make sense change the attribute to allow be used in `properties`, because today it targets `AttributeTargets.Parameter | AttributeTargets.Class` only.